packages/application-generic/src/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -273,29 +292,49 @@ export class TriggerEvent {
 
     for (const step of template?.steps) {
       const type = step.template?.type;
-      if (type) {
-        const channelType = STEP_TYPE_TO_CHANNEL_TYPE.get(type);
-
-        if (channelType) {
-          if (providers[channelType]) continue;
-          if (channelType === ChannelTypeEnum.IN_APP) {
-            providers[channelType] = InAppProviderIdEnum.Novu;
-          } else {
-            const provider = await this.getProviderId(
-              environmentId,
-              channelType
-            );
-            if (provider) {
-              providers[channelType] = provider;
-            }
-          }
+      if (!type) continue;
+
+      const channelType = STEP_TYPE_TO_CHANNEL_TYPE.get(type);
+      if (!channelType) continue;
+      if (providers[channelType]) continue;
```

Comment:
```suggestion
      if (!channelType || providers[channelType]) continue;
```
Good approach that we can even simplify it.